### ISSUE TITLE
[Bug]: mono assertion for Unity 2021.3.12f1 (and above?) IL2CPP games 

### ISSUE THREAD

**Issue by alextusinean** [LavaGang/MelonLoader#322]
### All of the following criteria must be met

- [X] All <a href="https://github.com/LavaGang/MelonLoader/blob/master/README.md#requirements">Requirements</a> must be installed.
- [X] Full ``Latest.log`` file included. If no file exists then leave this unchecked and state so.

### All of the following are optional to answer

- [X] Tried reinstalling the Game.
- [X] Tried reinstalling MelonLoader.
- [X] Tried restarting PC.
- [X] Was able to see the Start Screen.

### Describe the issue.

[MelonLoader_22-11-10_11-58-30.950.log](https://github.com/LavaGang/MelonLoader/files/9997980/MelonLoader_22-11-10_11-58-30.950.log)

The mono assertion is only printed on the console:
```
0
* Assertion: should not be reached at ..\mono\mini\tramp-amd64.c:229
```

### Did you attach your log file?

- [X] Yes, I attached my log file to the text box above.
- [ ] No, I could not find a log file at `{Game_Directory}\MelonLoader\Latest.log`

**Comment by spicebag** [comment]
This is also the case for me, apparently it's a case of a newer unity version braking Melonloader, hope it's fixed soon since I've got people asking me about the game that I'm modding for.

**Comment by obrua** [comment]
i got too,

**Comment by SamboyCoding** [comment]
This should be fixed with 0.6.0
